tests/helpers/stubs.py

```diff
@@ -359,6 +360,34 @@ def __init__(self, *valid_values):
         self.complete = lambda: [(val, '') for val in valid_values]
 
 
+class FakeStatusbarCommand(QObject):
+
+    """Stub for the statusbar command prompt."""
+
+    got_cmd = pyqtSignal(str)
+    clear_completion_selection = pyqtSignal()
+    hide_completion = pyqtSignal()
+    update_completion = pyqtSignal()
+    show_cmd = pyqtSignal()
+    hide_cmd = pyqtSignal()
+    textChanged = pyqtSignal()
+
+    def __init__(self, parent=None, name=None):
+        super().__init__(parent)
+        self._cursor_pos = 0
+        self._text = ""
+        self.cursorPosition = lambda: self._cursor_pos
+        self.text = lambda: self._text
+        self.prefix = lambda: self._text[0]
+        self.setFocus = lambda: None
```

Comment:
I was actually surprised those work and don't simply always return the things in the `__init__` scope - I guess they work because of [late binding closures](http://docs.python-guide.org/en/latest/writing/gotchas/#late-binding-closures).

Still I think I'd prefer having real methods for them to avoid confusion, even if it's a few lines more :wink: